test/test.js

```diff
@@ -51,17 +54,18 @@ describe('TodoMVC', function () {
       .evaluate(function () {
         return document.querySelectorAll('.todo-list li').length
       })
-      .then(function (todoListLength) {
+      .then(function (todoListLength, snapshotName) {
         // We start with an empty to-do list.
         todoListLength.should.eq(0)
 
         // Add a new to-do item.
         return nightmare.type('.new-todo', 'New fancy todo')
           .wait('.todo-list li')
+          .use(percySnapshot(snapshotName))
           .evaluate(function () {
             return document.querySelectorAll('.todo-list li').length
           })
-      })
+      }, this.test.fullTitle() /* for use as Percy snapshot name */)
```

Comment:
I wonder if it's possible for the `percySnapshot` call to get the `fullTitle` automatically? That would be ideal. I get the sense you may have already looked into this already though, @anaulin.

Reply:
Yeah, that would be ideal, but I don't know how to swing it. The `fullTitle` is only available in the context of the `mocha` test, so once we are inside the implementation of `percySnapshot(...)` we don't have access to it. We have access to some basic `nightmare` things in the implementation, because we have an instance of `nightmare`, but it doesn't know anything about `mocha`. That's at least my understanding of it.

I poked around for a while to see what was available in the context in which the `evaluate` part of the `percySnapshot(...)` implementation, but there doesn't seem to be anything related to the tests at all (which makes sense -- Nightmare doesn't know about Mocha). 🤷‍♀️

Reply:
So, long story short, anything we want to pass into the snapshot from the test context we have to pass explicitly, such as `this.test.fullTitle()`.